[iOS] Add testing script for iOS x86 build

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26632 [iOS] Add testing script for iOS x86 build**

### Summary

This script builds the TestApp (located in ios folder) to generate an iOS x86 executable via the `xcodebuild` toolchain on macOS. The goal is to provide a quick way to test the generated static libraries to see if there are any linking errors. The script can also be used by the iOS CI jobs. To run the script, simply see description below:

```shell
$ruby scripts/xcode_ios_x86_build.rb --help

-i, --install                    path to the cmake install folder
-x, --xcodeproj                  path to the XCode project file
```

### Note

The script mainly deals with the iOS simulator build. For the arm64 build, I haven't found a way to disable the Code Sign using the `xcodebuiild` tool chain (XCode 10). If anyone knows how to do that, please feel free to leave a comment below.

### Test Plan

- The script can build the TestApp and link the generated static libraries successfully
- Don't break any CI job

Differential Revision: [D17530990](https://our.internmc.facebook.com/intern/diff/D17530990)

xta0 committed:
[iOS] Add testing script for iOS x86 build

ljk53 commented on scripts/xcode_ios_x86_build.rb:
```diff
@@ -0,0 +1,52 @@
+require 'xcodeproj'
+
+options = {}
+option_parser = OptionParser.new do  |opts| 
```
not familiar with ruby style, but looks like there is extra space between 'do' and '|opts|'?

ljk53 commented on ios/TestApp/TestApp/ViewController.mm:
```diff
@@ -1,5 +1,5 @@
 #import "ViewController.h"
-#import <LibTorch/LibTorch.h>
```
are you going to delete the header file?

xta0 replied:
No, the umbrella header is installed via cocoapods, which is not needed in this case. I replaced it with `import<torch/script.h>`

xta0 committed:
Update on "[iOS] Add testing script for iOS x86 build"


### Summary

This script builds the TestApp (located in ios folder) to generate an iOS x86 executable via the `xcodebuild` toolchain on macOS. The goal is to provide a quick way to test the generated static libraries to see if there are any linking errors. The script can also be used by the iOS CI jobs. To run the script, simply see description below:

```shell
$ruby scripts/xcode_ios_x86_build.rb --help

-i, --install                    path to the cmake install folder
-x, --xcodeproj                  path to the XCode project file
```

### Note

The script mainly deals with the iOS simulator build. For the arm64 build, I haven't found a way to disable the Code Sign using the `xcodebuiild` tool chain (XCode 10). If anyone knows how to do that, please feel free to leave a comment below.

### Test Plan

- The script can build the TestApp and link the generated static libraries successfully
- Don't break any CI job
